### DISCUSSION TITLE
Support for PEP 723

### DISCUSSION THREAD

**Discussion by Ravencentric** [microsoft/pylance-release#6522] [Enhancement]
See: https://packaging.python.org/en/latest/specifications/inline-script-metadata/#inline-script-metadata
![image](https://github.com/user-attachments/assets/2a2b59c1-5611-466e-809a-57b4943dc952)

This will always be a valid TOML string and should be treated as such. My feature request is that VS Code should:
- Syntax highlight it as TOML
- Potentially format it too
- Create/manage/detect (?) venvs for PEP 723 scripts for development. Currently pylance will just complain that your pep 723 script is missing dependencies.

**Comment by TylerGrantSmith** [comment]
I would be curious if anyone stumbling upon this has any workarounds until support is added

**Reply by Ravencentric** [reply]
My current workaround is just creating a venv from the PEP 723 script. If you're using uv you can do `uv venv && uv export --script .\pep723.py | uv pip sync -`. This will create a `.venv` in the current directory which can then be picked up by vs code.

**Reply by albertotb** [reply]
Nice workaround. Is there a way to tell `uv` to use that `.venv` instead of creating a new temporary one?

**Reply by hugolundin** [reply]
My workaround ([suggested ](https://github.com/astral-sh/uv/issues/17667#issuecomment-3791299538)by @woodruffw ) is to to add the `environments-v2` directory to `python-env.globalSearchPaths`. Then you only need to run `uv sync --script script.py` to create the environment before it shows up in the "Python: Select Interpreter" menu: 

<img width="566" height="132" alt="image" src="https://github.com/user-attachments/assets/17ee4c21-4003-4035-846b-89a0461d5052" />

**Comment by jsmariegaard** [comment]
This is becoming very relevant with uv scripts https://docs.astral.sh/uv/guides/scripts/ - which are currently not easy to run or debug inside VS Code - but otherwise extremely powerful

**Reply by Ravencentric** [reply]
To add to this, pipx, hatch, and more tools already support PEP 723 and it's only getting more common (because it's a great feature 🙂).

**Comment by Deathnerd** [comment]
Bumping for visibility. This would be an excellent feature to have
